lib/engine/game/g_18_mo/game.rb

```diff
@@ -0,0 +1,247 @@
+                },
+                {
+                  name: '5',
+                  on: '5',
+                  train_limit: 3,
+                  tiles: %i[yellow green brown],
+                  operating_rounds: 2,
+                },
+                {
+                  name: '6',
+                  on: '6E',
+                  train_limit: 3,
+                  tiles: %i[yellow green brown],
+                  operating_rounds: 2,
+                },
+                {
+                  name: '7',
+                  on: '7',
+                  train_limit: 2,
+                  tiles: %i[yellow green brown gray],
+                  operating_rounds: 2,
+                },
+              ].freeze
+
+        TRAINS = [
+          {
+            name: '2Y',
+            distance: 2,
+            price: 100,
+            obsolete_on: '3E',
+            rusts_on: '7',
+            variants: [
+              {
+                name: '3Y',
+                distance: 3,
+                price: 150,
+              },
+              ],
+          },
+          {
+            name: '3G',
+            distance: 3,
+            price: 180,
+            obsolete_on: '5E',
+            variants: [
+              {
+                name: '4G',
+                distance: 4,
+                price: 240,
+              },
+            ],
+          },
+          {
+            name: '3E',
+            distance: [{ 'nodes' => %w[city offboard town], 'pay' => 3, 'visit' => 99 }],
+            price: 300,
+            obsolete_on: '6E',
+          },
+          {
+            name: '5',
+            distance: 5,
+            price: 450,
+            variants: [
+              {
+                name: '6',
+                distance: 6,
+                price: 540,
+              },
+            ],
+            events: [{ 'type' => 'close_companies' }],
+          },
+          {
+            name: '5E',
+            distance: [{ 'nodes' => %w[city offboard town], 'pay' => 5, 'visit' => 99 }],
+            price: 600,
+          },
+          {
+            name: '7',
+            distance: 7,
+            price: 700,
+            variants: [
+              {
+                name: '8',
+                distance: 8,
+                price: 800,
+              },
+            ],
+            events: [
+              { 'type' => 'remove_markers' },
+              { 'type' => 'remove_reservations' },
+            ],
+          },
+        ].freeze
+
+        def operating_order
+          @minors.select(&:floated?) + @corporations.select(&:floated?).sort
+        end
+
+        def check_other(route)
+          visited_hexes = {}
+          route.visited_stops.each do |stop|
+            hex = stop.hex
+            raise GameError, 'Route cannot run to multiple cities in a hex' if visited_hexes[hex]
+
+            visited_hexes[hex] = true
+          end
+        end
+
+        # Two lays with one being an upgrade, second tile costs 20
+        TILE_LAYS = [
```

Comment:
this shouldn't be here, it should be above

Reply:
ok